src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobWorker.cs

```diff
@@ -69,16 +72,31 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
                         }
                     }
 
-                    await Task.Delay(_exportJobConfiguration.JobPollingFrequency, cancellationToken);
-                }
-                catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
-                {
-                    // Cancel requested.
+                    // We successfully completed an attempt to acquire export jobs. Let us reset the polling frequency in case it has changed.
+                    delayBeforeNextPoll = _exportJobConfiguration.JobPollingFrequency;
                 }
                 catch (Exception ex)
                 {
                     // The job failed.
                     _logger.LogError(ex, "Unhandled exception in the worker.");
+
+                    // Since acquiring jobs failed let us introduce a delay before we retry. We don't want to increase the delay between polls to more than an hour.
+                    delayBeforeNextPoll *= 2;
+                    if (delayBeforeNextPoll > _maximumDelay)
+                    {
+                        delayBeforeNextPoll = _maximumDelay;
+                    }
+                }
+
+                try
+                {
+                    await Task.Delay(delayBeforeNextPoll, cancellationToken);
+                }
+                catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
```

Comment:
Is there a chance a cancellation could be requested during the upper try block? It looks like it would be handled in the upper catch block and then would continue looping.

Reply:
Even in that scenario, the `await Task.Delay()` will throw a `TaskCanceledException` which will then be caught by the `catch (OperationCanceledException)` block. The while loop condition also checks for whether cancellation was requested (`while (!cancellationToken.IsCancellationRequested)`). So we will exit the loop one way or the other.